app.js

```diff
@@ -108,44 +131,71 @@ router.post('/account/:phoneNumber/:accountId/requestCode', async ctx => {
 const nacl = require('tweetnacl');
 const crypto = require('crypto');
 const bs58 = require('bs58');
-const verifySignature = async (nearAccount, securityCode, signature) => {
-    const hasher = crypto.createHash('sha256');
-    hasher.update(securityCode);
-    const hash = hasher.digest();
-    const helperPublicKey = (await keyStore.getKey(recoveryKeyJson.account_id)).publicKey;
-    const accessKeys = await nearAccount.getAccessKeys();
-    if (!accessKeys.find(it => it.public_key == helperPublicKey.toString())) {
-        throw Error(`Account ${nearAccount.accountId} doesn't have helper key`);
+const verifySignature = async (nearAccount, data, signedData) => {
+    try {
+        const hash = crypto.createHash('sha256').update(data).digest();
+        const accessKeys = await nearAccount.getAccessKeys();
+        return accessKeys.some(it => {
+            const publicKey = it.public_key.replace('ed25519:', '');
+            return nacl.sign.detached.verify(hash, Buffer.from(signedData, 'base64'), bs58.decode(publicKey));
+        });
+    } catch (e) {
+        console.error(e);
```

Comment:
hmm, are there any expected errors? Shouldn't we just fail through normal failure path (i.e. 500) for unexpected errors?

Reply:
If you provide a garbage signature, then the `verify` line throws an error. This was happening in the tests a fair bit at one point (not sure if it still would), and could potentially happen in the wild as well. If that happens, we want this function to return `false`. Do you agree?

Reply:
either one is ok from my POV. I'd probably just have let it throw out of laziness, but should be fine as is as well.